### PR TITLE
Update to towncrier 22.12.0, and other fixes

### LINKED ISSUE #16: Incompatibility with towncrier 22.12.0
Started seeing the error that @namurphy reported at https://github.com/twisted/towncrier/issues/461 . Something something about "abusing private API" over there.

### PR DESCRIPTION
**Pull request by dstansby**
Fixes https://github.com/OpenAstronomy/sphinx-changelog/issues/16

### CODE REVIEW

File: sphinx_changelog/towncrier.py [hunk 1]
```diff
@@ -15,6 +15,25 @@
...
 
 
+class _Config:
+    """
+    Wrapper class to make the (private) config object in towncrier
+    subscriptable.
+
+    In version 22.12.0 the config object was turned
+    from a dict to a dataclass, so this can be removed if/when
+    towncrier is pinned to >= 22.12.
```
Comment (Cadair) [inline sphinx_changelog/towncrier.py]: I have been thinking about towncrier releases and this package.

My thinking is that we should hard pin a single release of towncrier for this package and update it explicitly every time. My reasoning for this unusual decision is two fold:

1) We are relying on a private unstable API of towncrier, which they make no promises about and therefore break regularly.
1) This is not library package it's a sphinx extension, only needed for doc builds, which 99.99% of the time will have no other need for towncrier.

The only potential issue I can see is if towncrier release a change which breaks the config, packages will need to keep this and the towncrier versions in sync, but I don't think that should be an issue as long as we keep up with tc releases.

So, all that being said, I suggest you update the code to rely only on 22.12 and hard pin it in setup.cfg.

File: tox.ini [hunk 2]
```diff
@@ -1,6 +1,5 @@
 [tox]
 envlist =
-    py{36,37,38}-test
```
Comment (Cadair) [inline tox.ini:3]: The diff of this file seems wonky? you have deleted all the actual test lines no?

Comment (dstansby) [reply tox.ini:3]: Yeah, that was deliberate - the test environments doesn't do anything, as there aren't any tests to run. Maybe I'm naïve for thinking test environments defined in `tox.ini` files should do something, but I think it's clearer to a new dev running `tox -l` to only have useful and working environments defined.

**Review by dstansby — COMMENTED** [review]
Pinned `towncrier`. I also added a new test environment that tests with the latest version of towncrier so it's easy to check future compatability.

**Review by Cadair — APPROVED** [review]

**Comment by pllim** [comment]
Thanks!

**Comment by pllim** [comment]
Any chance `sphinx-changelog` can get a new release with this patch, so we can unpin towncrier downstream? Thanks again!

**Comment by Cadair** [comment]
Yeah, I need to find time to fix the CI and then I can do a release. It's on [the list](https://github.com/users/Cadair/projects/2/views/1)

**Pull request merged**